src/System.Management.Automation/engine/CommandCompletion/CompletionCompleters.cs

```diff
@@ -4400,350 +4405,662 @@ internal static IEnumerable<CompletionResult> CompleteFilename(CompletionContext
+
+                    if (!inputUsedHomeChar && basePath[0] is not '/' and not '\\')
+                    {
+                        defaultRelativePath = !context.ExecutionContext.LocationGlobber.IsAbsolutePath(wordToComplete, out _);
+                    }
                 }
+            }
 
-                var powerShellExecutionHelper = context.Helper;
-                powerShellExecutionHelper
-                    .AddCommandWithPreferenceSetting("Microsoft.PowerShell.Management\\Resolve-Path")
-                    .AddParameter("Path", wordToComplete + "*");
+            StringConstantType stringType;
+            if (quote.Equals(string.Empty, StringComparison.Ordinal))
+            {
+                stringType = StringConstantType.BareWord;
+            }
+            else if (quote.Equals("\"", StringComparison.Ordinal))
+            {
+                stringType = StringConstantType.DoubleQuoted;
+            }
+            else
+            {
+                stringType = StringConstantType.SingleQuoted;
+            }
 
-                Exception exceptionThrown;
-                var psobjs = powerShellExecutionHelper.ExecuteCurrentPowerShell(out exceptionThrown);
+            var useLiteralPath = context.GetOption("LiteralPaths", @default: false);
+            if (useLiteralPath)
+            {
+                basePath = EscapePath(basePath, stringType, useLiteralPath, out _);
+            }
 
-                if (psobjs != null)
-                {
-                    var isFileSystem = false;
-                    var wordContainsProviderId = ProviderSpecified(wordToComplete);
+            _ = context.Helper
+                .AddCommandWithPreferenceSetting("Microsoft.PowerShell.Management\\Resolve-Path")
+                .AddParameter("Path", basePath);
+
+            var resolvedPaths = context.Helper.ExecuteCurrentPowerShell(out _);
+            if (resolvedPaths is null || resolvedPaths.Count == 0)
+            {
+                return CommandCompletion.EmptyCompletionResult;
+            }
+
+            var useRelativePath = context.GetOption("RelativePaths", @default: defaultRelativePath);
+            if (useRelativePath && providerSeparatorIndex != -1)
+            {
+                // User must have requested relative paths but that's not valid with provider paths.
+                return CommandCompletion.EmptyCompletionResult;
+            }
+
+            var resolvedProvider = ((PathInfo)resolvedPaths[0].BaseObject).Provider;
+            string providerPrefix;
+            if (providerSeparatorIndex == -1)
+            {
+                providerPrefix = string.Empty;
+            }
+            else if (providerSeparatorIndex == resolvedProvider.Name.Length)
+            {
+                providerPrefix = $"{resolvedProvider.Name}::";
+            }
+            else
+            {
+                providerPrefix = $"{resolvedProvider.ModuleName}\\{resolvedProvider.Name}::";
+            }
```

Comment:
The same.

Reply:
Not possible. The switch expects a constant value.